kie-soup-project-datamodel/kie-soup-project-datamodel-commons/src/main/java/org/kie/soup/project/datamodel/commons/oracle/ModuleDataModelOracleImpl.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.kie.soup.project.datamodel.commons.oracle;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.kie.soup.project.datamodel.oracle.Annotation;
+import org.kie.soup.project.datamodel.oracle.MethodInfo;
+import org.kie.soup.project.datamodel.oracle.ModelField;
+import org.kie.soup.project.datamodel.oracle.ModuleDataModelOracle;
+import org.kie.soup.project.datamodel.oracle.TypeSource;
+
+/**
+ * Default implementation of DataModelOracle
+ */
+public class ModuleDataModelOracleImpl implements ModuleDataModelOracle {
+
+    //Module name
+    protected String moduleName;
+
+    //Fact Types and their corresponding fields
+    protected Map<String, ModelField[]> moduleModelFields = new HashMap<String, ModelField[]>();
+
+    //Map of the field that contains the parametrized type of a collection
+    //for example given "List<String> name", key = "name" value = "String"
+    protected Map<String, String> moduleFieldParametersType = new HashMap<String, String>();
```

Comment:
I realize this PR is just renaming, but since we're touching this code, I think it would be better if these explanatory comments were in "public" javadoc on ModuleDataModelOracle interface instead of hidden here, as they eplain the semantics of data kept by the oracle..

Reply:
Personally, I think fixing this is scope creep that should be avoided. The related PRs to appformer and kie-wb-common have ~1000 changed files combined, and there are lots of known issues that could be worked on.

Reply:
I agree with @mbarkley These PRs have been in flux for >3 months (if not a lot longer).. the sooner we get the required changes merged the sooner we can move on.... including improving code not refactored in these PRs (feel free to create a JIRA and assign to me if you like @jhrcek)

Reply:
@mbarkley @manstis taken to heart. I don't want to introduce unnecessary delay. I'll keep my review comments to the point.